locked-asset/energy-factory/src/unlocked_token_transfer.rs

```diff
@@ -0,0 +1,55 @@
+multiversx_sc::imports!();
+
+#[multiversx_sc::module]
+pub trait UnlockedTokenTransferModule:
+    utils::UtilsModule
+    + multiversx_sc_modules::pause::PauseModule
+    + crate::token_whitelist::TokenWhitelistModule
+{
+    #[only_owner]
+    #[endpoint(addToUnlockedTokenTransferWhitelist)]
+    fn add_to_unlocked_token_transfer_whitelist(
+        &self,
+        sc_addresses: MultiValueEncoded<ManagedAddress>,
+    ) {
+        let mapper = self.unlocked_token_transfer_whitelist();
+        for sc_addr in sc_addresses {
+            self.require_sc_address(&sc_addr);
+
+            mapper.add(&sc_addr);
+        }
+    }
+
+    #[only_owner]
+    #[endpoint(removeFromUnlockedTokenTransferWhitelist)]
+    fn remove_from_unlocked_token_transfer_whitelist(
+        &self,
+        sc_addresses: MultiValueEncoded<ManagedAddress>,
+    ) {
+        let mapper = self.unlocked_token_transfer_whitelist();
+        for sc_addr in sc_addresses {
+            mapper.remove(&sc_addr);
+        }
+    }
+
+    #[endpoint(transferUnlockedToken)]
+    fn transfer_unlocked_token(&self, dest: ManagedAddress, amount: BigUint) {
```

Comment:
The caller should not be given as a parameter. We should have an only_owner endpoint which sets the address in the energy factory storage. And we use only that. The farm should not have a saying in where the tokens are sent.

Reply:
Disagree. Not only farms will potentially use this, and they don't all have the same owner. We've discussed this previously.